### PR TITLE
perf: reduce dropConf

### PR DESCRIPTION
**Pull request by noyyyy**

### CODE REVIEW

File: src/lib/SingleRanking.sol [hunk 1]
```diff
@@ -70,41 +70,56 @@ library SingleRanking {
...
+                unchecked {
+                    _offset -= FastArray.length(keys);
+                }
             } else if (FastArray.length(keys) < _offset + _count) {
-                uint index = FastArray.length(keys) - 1;
-
+                uint index;
+                unchecked {
+                    index = FastArray.length(keys) - 1;
+                }
```
Comment (yuanjunliang) [inline src/lib/SingleRanking.sol:80]: delete unchecked, will overflow when fastArray.length = 0

File: src/lib/SingleRanking.sol [hunk 2]
```diff
@@ -70,41 +70,56 @@ library SingleRanking {
...
+                        size += 1;
+                    }
                     if (index == 0) {
                         break;
                     }
 
-                    index -= 1;
+                    unchecked {
+                        index -= 1;
+                    }
```
Comment (yuanjunliang) [inline src/lib/SingleRanking.sol:94]: delete unchecked, will overflow if index and offset eq 0.

File: src/lib/SingleRanking.sol (hunk 1 above)
Comment (noyyyy) [reply src/lib/SingleRanking.sol:80]: If `FastArray.length(keys)` equals zero, `_offset >= FastArray.length(keys)` is true and this branch will be not executed.

File: src/lib/SingleRanking.sol (hunk 2 above)
Comment (noyyyy) [reply src/lib/SingleRanking.sol:94]: If index equals zero, the while loop will stop and exit before `index -=1`

**Review by yuanjunliang — APPROVED** [review]

**Review by yuanjunliang — APPROVED** [review]

**Pull request merged**